rpc-tests/tests/test-rpc.js

```diff
@@ -21,69 +25,80 @@ describeWithNetwork(network, `${network} RPC`, function(context) {
 		expect(chain.toString()).to.equal(`${capitalize(network)} Testnet`);
 	});
 
-	it('should fetch chain name from rpc node', async function () {
+	it('should fetch chain name from rpc node', async () => {
 		const name = await context.api.rpc.system.name();
 
 		expect(name.toString()).to.equal('Astar Collator');
 	});
 
-	it('should be able to Register contract on H160 address 0x01 using Alice account', async function () {
-        const api = context.api;
-
-        // Construct the keyring after the API (crypto has an async init)
-        const keyring = new Keyring({ type: 'sr25519' });
-      
-        // Add Alice to our keyring with a hard-derivation path (empty phrase, so uses dev)
-        const alice = keyring.addFromUri('//Alice');
-
-        // Create a extrinsic, transferring 100 units to Bob
-        const transfer = api.tx.dappsStaking.register(getAddressEnum(CONTRACT));
-      
-        // Sign and send the transaction using our account
-        const hash = await transfer.signAndSend(alice, { nonce: -1 });
-
-        console.log('Transfer sent with hash', hash.toHex());
-
-        expect(hash).to.exist;
+	it('should be able to Register contract on H160 address 0x01 using Alice account', async () => {
```

Comment:
It looks to me like we are testing if a node fired correct event with correct arguments. I believe that in our case we need to compare states before and after transaction. That change will make tests more readable and more concise.

For example if you change sendTransaction method to return boolean when transaction is finalized (when transaction is finalized you can only look for ExtrinsicFailed event so you can return false, which should stop test execution), you can rewrite tests as below (also you can omit some parameters from endTransaction call).

<img width="702" alt="image" src="https://user-images.githubusercontent.com/8452361/181239133-135a281d-ee78-4c16-9367-fc54dddfb555.png">

Reply:
Done